packages/compass-sidebar/src/components/use-filtered-connections.ts

```diff
@@ -388,16 +388,19 @@ function filteredConnectionsToSidebarConnection(
   return sidebarConnections;
 }
 
+export type ConnectionsFilter = {
+  regex: RegExp | null;
```

Comment:
Nit and not something you changed, but I'm wondering why this ever used explicit null over just being optional / undefined.

Reply:
It does 👍 https://github.com/mongodb-js/compass/blob/28bb67513c1eca57ff14ced8d0c01562452a40a7/packages/compass-sidebar/src/components/navigation-items-filter.tsx#L23-L27